onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2006,10 +2020,29 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<Node*>& fuse
         }
       }
 
-      // Run TRT inference
-      if (!trt_context->enqueueV2(&buffers[0], stream, nullptr)) {
-        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "TensorRT EP execution context enqueue failed.");
-      }
+      // Run TRT inference			  
+      if (trt_state->cuda_graph_enable)
+      {
+          if (*cuda_graph == nullptr) {
```

Comment:
i think this entire path isn't thread safe as cuda graphs and associated api's don't seem to be thread safe.

Reply:
Yes, if cuda graphs are enabled, then Run() will no longer be thread-safe and calls to Run() needs to be serialized either by the caller or ORT itself before replaying the graph.

Reply:
we have lock in the inference and the compute() is serialized already.

Reply:
ah yes tensorrt_mu_ already protects the compute_func body.